core/shared/src/main/scala/zio/internal/metrics/MetricState.scala

```diff
@@ -0,0 +1,70 @@
+package zio.internal.metrics
+
+import zio._
+import zio.internal.ZMetrics._
+
+final case class MetricState(
+  name: String,
+  help: String,
+  labels: Chunk[Label],
+  details: MetricType
+) {
+  override def toString(): String = {
+    val lbls = if (labels.isEmpty) "" else labels.map(l => s"${l._1}->${l._2}").mkString("{", ",", "}")
+    s"MetricState($name$lbls, $details)"
+  }
+}
+
+object MetricState {
+
+  // --------- Methods creating and using Prometheus counters
+  def counter(key: MetricKey.Counter, help: String, value: Double): MetricState =
+    MetricState(key.name, help, Chunk(key.tags: _*), MetricType.Counter(value))
+
+  // --------- Methods creating and using Prometheus Gauges
+
+  def gauge(
+    key: MetricKey.Gauge,
+    help: String,
+    startAt: Double
+  ): MetricState =
+    MetricState(key.name, help, Chunk(key.tags: _*), MetricType.Gauge(startAt))
```

Comment:
Performance note but probably `key.tags` should already be a chunk so we don't have to wrap it every time.

Reply:
👍